### ISSUE TITLE
Addition Apple M1 Pro Issues

### ISSUE THREAD

**Issue by stefanpenner** [sebhildebrandt/systeminformation#660]
**Describe the bug**
It appears that this library has some additional issues when running on a machined powered by Apple M1 Pro.

[`modelline`](https://github.com/sebhildebrandt/systeminformation/blob/57c7d9af587435a3c626817be7e27617f451e397/lib/cpu.js#L610-L612)
 ends up being 'Apple M1 Pro' on a new Mac Book Pro,  which never contains an `@`.. this means the result of  `modelline.split('@')` will be null, causing https://github.com/sebhildebrandt/systeminformation/blob/57c7d9af587435a3c626817be7e27617f451e397/lib/cpu.js#L612 to fail with when attempting to index into `null`.
  
 Machine Details:

**Environment (please complete the following information):**
 <img width="235" alt="image" src="https://user-images.githubusercontent.com/1377/156244322-f4f3653c-1284-4015-8614-8e0ff2b1ba90.png">
* Node 14.17.5

**Comment by sebhildebrandt** [comment]
@stefanpenner ... hmmm interesting, as my main machine is a Apple M1 Pro machine ... I cannot reproduce the error:

In file `cpu.js`
```
const modelline = util.getValue(lines, 'machdep.cpu.brand_string');
```

indeed returns `Apple M1 Pro`

and the next line 

```
const modellineParts = modelline.split('@');
```

retuns `[ 'Apple M1 Pro' ]`

so the next 2 following lines are also fine:

```
result.brand = modellineParts[0].trim();
const speed = modellineParts[1] ? modellineParts[1].trim() : '0';
```

The array has only the length of 0 but this is covered by checking if `modellineParts[1]` exists ...

With the current version I get the following result:

```
{
  manufacturer: 'Apple',
  brand: 'M1 Pro',
  vendor: 'Apple',
  family: '458787763',
  model: '',
  stepping: '4',
  revision: '',
  voltage: '',
  speed: 2.4,
  speedMin: 2.4,
  speedMax: 2.4,
  governor: '',
  cores: 10,
  physicalCores: 10,
  processors: 1,
  socket: 'SOC',
  flags: '',
  virtualization: false,
  cache: { l1d: 131072, l1i: 65536, l2: 4194304, l3: null },
  efficiencyCores: 2,
  performanceCores: 8
}
```

Did I miss something? I do not see anything failing. Maybe I understood your message ...

**Comment by sebhildebrandt** [comment]
@stefanpenner ... as I cannot reproduce this on any of my M1 machines I will close this for now (see explanation above). If you still see problems, please reopen this issue and provide a little more error messages and information. Thank you so much.

**Comment by sebhildebrandt** [comment]
@stefanpenner sure. Whenever you have more info, feel free to reopen this issue. I am highly interested in fixing issues and make this library as stable as possible. Best regards.

**Comment by thoov** [comment]
While testing the issue that @stefanpenner found I was able to reproduce this error.

<img width="1000" alt="Screen Shot 2022-03-11 at 1 40 09 PM" src="https://user-images.githubusercontent.com/166909/157974211-a1ffb8a8-fda4-4595-a5c6-1bf05f51f489.png">

Looking at where the error is thrown I see:

```
// ...
const modelline = util.getValue(lines, 'machdep.cpu.brand_string');
result.brand = modelline.split('@')[0].trim();
result.speed = modelline.split('@')[1].trim();
// ...
```

Which  as `modelline` for an M1 is `Apple M1 Pro` that second split will not work. This is fixed in newer versions as we were on v4.21.2 (https://github.com/sebhildebrandt/systeminformation/blob/v4.21.2/lib/cpu.js#L415-L417)

**Comment by stefanpenner** [comment]
Thanks, @thoov clearly, I got myself turned around thinking i was fully upgraded when I wasn't. My bad.
